src/java.base/share/classes/sun/nio/ch/FileChannelImpl.java

```diff
@@ -585,6 +585,17 @@ private long transferToTrustedChannel(long position, long count,
         if (!((target instanceof FileChannelImpl) || isSelChImpl))
             return IOStatus.UNSUPPORTED;
 
+        if (target == this) {
+            synchronized (positionLock) {
+                long posThis = position();
+                if (posThis - count + 1 <= position &&
```

Comment:
I realise I suggested we need the positionLock here but I don't think it is needed now because there is only one call to position().

Reply:
Will remove.